src/sentry/integrations/github/integration.py

```diff
@@ -199,16 +199,16 @@ def get_pipeline_views(self):
         return [GitHubInstallationRedirect()]
 
     def get_installation_info(self, installation_id):
-        session = http.build_session()
         headers = {
             # TODO(jess): remove this whenever it's out of preview
             "Accept": "application/vnd.github.machine-man-preview+json",
         }
         headers.update(jwt.authorization_header(get_jwt()))
-        resp = session.get(
-            "https://api.github.com/app/installations/%s" % installation_id, headers=headers
-        )
-        resp.raise_for_status()
+        with http.build_session() as session:
+            resp = session.get(
+                f"https://api.github.com/app/installations/{installation_id}", headers=headers
+            )
+            resp.raise_for_status()
         installation_resp = resp.json()
```

Comment:
does that work? It wouldn't make sense to me to start reading response after you closed the socket...

Reply:
Yeah, it works. It isn't lazy like that.